File: Python/1260.py
result = ""
visit = []
graph = []


def dfs(V):
    global result
    result += str(V) + " "
    visit[V] = 1  # 방문 표시

    for node in graph[V]:
        if visit[node] == 0:  # 해당 정점을 아직 방문하지 않았으면
            dfs(node)


queue = []


def bfs(V):
    global result
    queue.append(V)
    visit[V] = 1

    while len(queue) != 0:
        vertex = queue.pop(0)
        result += str(vertex) + " "
        for node in graph[vertex]:
            if visit[node] == 0:
                queue.append(node)
                visit[node] = 1


if __name__ == "__main__":
    N, M, V = map(int, input().split())
    graph = [[] for _ in range(N + 1)]
    visit = [0 for _ in range(N + 1)]

    for _ in range(M):
        start, end = map(int, input().split())
        graph[start].append(end)
        graph[end].append(start)

    for i in range(len(graph)):
        graph[i] = sorted(graph[i])

    dfs(V)
    result += "\n"
    visit = [0 for _ in range(N + 1)]
    bfs(V)
    
    print(result)

File: C++/1260.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <queue>
using namespace std;
typedef int num;

static vector<vector<num>> A;
static vector<bool> visited;

void dfs(num node) {
	cout << node << " ";
	visited[node] = true;
	for (num i : A[node]) {
		if (!visited[i]) {
			dfs(i);
		}
	}
}
void bfs(num node) {
	queue<int> q;
	q.push(node);
	visited[node] = true;
	while (!q.empty()) {
		num current = q.front();
		q.pop();
		cout << current << " ";
		for (num i : A[current]) {
			if (!visited[i]) {
				visited[i] = true;
				q.push(i);
			}
		}
	}
}
int main()
{
	ios_base::sync_with_stdio(false);
	cout.tie(0);
	cin.tie(0);

	num n, m, v;
	cin >> n >> m >> v;
	A.resize(n + 1);
	for (num i = 0; i < m; i++) {
		num s, e;
		cin >> s >> e;
		A[s].push_back(e);
		A[e].push_back(s);
	}
	for (num i = 1; i <= n; i++) {
		sort(A[i].begin(), A[i].end());
	}

	visited = vector<bool>(n + 1, false);
	dfs(v);
	cout << "\n";
	fill(visited.begin(), visited.end(), false);
	bfs(v);
	cout << "\n";
}